fw/procfs.c

```diff
@@ -359,7 +402,57 @@ tfw_procfs_stop(void)
 	tfw_procfs_cleanup();
 }
 
+static int
+tfw_cfgop_health_stat(TfwCfgSpec *cs, TfwCfgEntry *ce)
+{
+	int cpu;
+
+	TFW_CFG_CHECK_VAL_N(>, 0, cs, ce);
+	TFW_CFG_CHECK_NO_ATTRS(cs, ce);
+
+	health_stat_codes = kzalloc(tfw_hm_stats_size(ce->val_n),
+				    GFP_KERNEL);
+	if (!health_stat_codes)
+		return -ENOMEM;
+	if (tfw_hm_stats_init_from_cfg_entry(health_stat_codes, ce)) {
+		kfree(health_stat_codes);
+		health_stat_codes = NULL;
+		return -EINVAL;
```

Comment:
We can just return -EINVAL, `health_stat_codes` will be freed in the `tfw_cfgop_cleanup_health_stat`